routes/conversation.js

```diff
@@ -255,6 +257,74 @@ exports.leave = function (req, res){
 	});
 };
 
+/**
+ * Action when a user archives a conversation.
+ */
+exports.archive = function (req, res){
+	if (!req.user || !req.user._id){
+		resError(res, "Access denied.", "/error");
+		return;
+	}
+
+	// Search through the user's conversation list until the conversation with the correct
+	// id is found. 
+	User.findById( req.user._id, function (err, user){
+		if (err){
```

Comment:
So, I found an error that's my fault, and I only realized it today.  When we do a database query and the results are empty, the query actually doesn't seem to throw an error.  Instead, it simply runs the callback with a null result.  For example, in this `User.findById` example, when a user is not found, err == null and user == null.

Therefore, everywhere we have `if (err)` and we really mean "if there was an error or the result is empty", we should really be writing `if (err || !user)`.  This is really my fault, I did not understand what the `err` variable really meant, I only assumed it included empty queries when it did not.  I'm sorry for making that assumption.  I'll be changing a lot of these instances in my PR, as well.  Could you please change this `if (err)` to `if (err || !user)`?